docs/developer-docs/smart-contracts/compile.mdx

```diff
@@ -0,0 +1,75 @@
+---
+keywords: [beginner, tutorial, compile]
+---
+
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+import { GlossaryTooltip } from "/src/components/Tooltip/GlossaryTooltip";
+
+# Compile
+
+<MarkdownChipRow labels={["Beginner", "Tutorial"]} />
+
+## Overview
+
+After you have written the code for your project's <GlossaryTooltip>canisters</GlossaryTooltip>, you need to compile the code into a WebAssembly module before it can be deployed on ICP. 
+
+## How code is compiled to Wasm
+
+WebAssembly (Wasm) is platform independent and designed to run on a stack-based virtual machine. Wasm itself is not a programming language; it is the result of a programming language being comnpiled and converted into executable machine code. When canister code is compiled into Wasm, the system compiler creates a Wasm module in either a binary format (.wasm) or a compressed binary (.wasm.gz).
+
+This generated Wasm module includes a runtime system which implements a garbage collector function and uses the Wasm heap storage. Then, this module is implemented when the canister's code is installed. 
+
+<div class="text--center">
+<img src="/img/docs/wasm-module.png" alt="Compile Wasm" width="600"/>
+</div>
+
+On ICP, [Wasmtime](https://docs.wasmtime.dev/) is supported to run canister smart contracts. 
+
+## Compiling your canister
+
+To compile your canister's Wasm module, you can use the `dfx build` command. The `dfx build` command looks for source code to compile for each canister configured under the `canisters` section in the `dfx.json` configuration file. It can be used to compile a specific canister or all canisters defined in the project's [`dfx.json`](/docs/current/developer-docs/developer-tools/cli-tools/dfx-json-reference) file.
+
+Check out the [default project template](/docs/current/developer-docs/getting-started/default-template) for more information on `dfx.json`.
+
+Before using the `dfx build` command, first you need to [create the canister](/docs/current/developer-docs/smart-contracts/create) and verify the location of your project's files and their file names. If you modify these settings, be sure to edit your project's `dfx.json` file to reflect the current intended configuration.
+
+Then, [compile your canister code](/docs/current/developer-docs/developer-tools/cli-tools/cli-reference/dfx-build) with the command:
+
+```
+dfx build // build locally
+dfx build --network ic // build on the mainnet
+```
+
+:::caution
+Compiling a canister on the mainnet will cost [cycles](/docs/current/developer-docs/getting-started/cycles/cycles-faucet).
```

Comment:
i'm a bit confused by this statement. compiling is done locally, but deployment is what costs cycles, correct?

Reply:
Compilation happens on the local machine of the developer.
Compiled Wasm binary is then installed on the target canister, which costs cycles.

Reply:
Correct, I've revised for clarity.